packages/transformer/src/IModelTransformer.ts

```diff
@@ -755,6 +772,27 @@ export class IModelTransformer extends IModelExportHandler {
    */
   public override shouldExportElement(_sourceElement: Element): boolean { return true; }
 
+  public override onSkipElement(sourceElement: Element): void {
+    if (this.context.findTargetElementId(sourceElement.id) !== Id64.invalid) {
+      // element already has provenance
+      return;
+    }
+
+    Logger.logInfo(loggerCategory, `Element '${sourceElement.id}' won't be exported. Marking its references as resolved`);
+    // Remap skipped element to non-existent EcInstanceId.
+    // When exporting other elements transformer needs reset all references to missingReferenceId
+    this.context.remapElement(sourceElement.id, missingReferenceId);
```

Comment:
This is a bit too hacky... and imo not future proof. Not likely but that 0x2 id could eventually be taken or elsehow made invalid for usage. We really need to move that native element remap table into TypeScript. I have some initial work for it but because of how geometry is processed in the native layer, it's not simple, so I haven't finished it.

I think it's possible to instead for now introduce a set of missing reference ids that must be checked in onTransformBlah and maybe before creating pending references. Can you try that instead?

Reply:
Added skipped entities property for tracking entities that are not exported. Although for now made it only track elements since it should bring the most benefit as most relationships are between elements. Also solution seems cleaner since actually checks onTransform___ are no longer needed since current onTransform implementations already handle cases when reference is missing. The only thing that was needed to be added was additonal check so that pending references wouldn't be added to already skipped elements.